From 7d66a76b608e1945a6363188afe0b1320e2fd70f Mon Sep 17 00:00:00 2001
From: "deepak.s" <deepak.s@adlinktech.com>
Date: Tue, 27 Aug 2024 11:46:15 +0530
Subject: [PATCH] Added DSI panel support

---
 drivers/video/mediatek/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: git/drivers/video/mediatek/Makefile
===================================================================
--- git.orig/drivers/video/mediatek/Makefile
+++ git/drivers/video/mediatek/Makefile
@@ -22,7 +22,7 @@ ifeq ($(CONFIG_VIDEO_MEDIATEK),y)
 			dp_mt8188.o \
 			vdosys_mt8188.o \
 			disp_pwm_mt8188.o
-		obj-$(CONFIG_TARGET_MT8195) += \
+		obj-$(CONFIG_TARGET_LEC_MTK_I1200) += \
 			dsi_mt8195.o \
 			dp_mt8195.o \
 			vdosys_mt8195.o \
Index: git/include/configs/lec_mtk_i1200.h
===================================================================
--- git.orig/include/configs/lec_mtk_i1200.h
+++ git/include/configs/lec_mtk_i1200.h
@@ -102,21 +102,27 @@
 	BOOT_TARGET_USB(func) \
 	BOOT_TARGET_SCSI(func)
 
-#if !defined(CONFIG_EXTRA_ENV_SETTINGS)
+#if defined(CONFIG_ENV_IS_IN_SPI_FLASH)
+#define SPLASH_SOURCE_SETTING "splashsource=sf\0"
+#else
+#define SPLASH_SOURCE_SETTING "splashsource=scsi_fs\0"
+#endif
+
+#ifndef CONFIG_EXTRA_ENV_SETTINGS
 #define CONFIG_EXTRA_ENV_SETTINGS \
-	"scriptaddr=0x40000000\0" \
-	"fdt_addr_r=0x44000000\0" \
-	"fdtoverlay_addr_r=0x44c00000\0" \
-	"kernel_addr_r=0x45000000\0" \
-	"ramdisk_addr_r=0x46000000\0" \
-	"bootargs=net.ifnames=0\0" \
-	"fdtfile=" CONFIG_DEFAULT_DEVICE_TREE ".dtb\0" \
-	"splashimage=" __stringify(CONFIG_SYS_LOAD_ADDR) "\0" \
-	"splashsource=scsi_fs\0" \
-	"splashfile=logo.bmp\0" \
-	"splashdevpart=2#bootassets\0" \
-	"splashpos=m,m\0" \
-	BOOTENV
+        "scriptaddr=0x40000000\0" \
+        "fdt_addr_r=0x44000000\0" \
+        "fdtoverlay_addr_r=0x44c00000\0" \
+        "kernel_addr_r=0x45000000\0" \
+        "ramdisk_addr_r=0x46000000\0" \
+        "bootargs=net.ifnames=0\0" \
+        "fdtfile=" CONFIG_DEFAULT_DEVICE_TREE ".dtb\0" \
+        "splashimage=" __stringify(CONFIG_SYS_LOAD_ADDR) "\0" \
+        SPLASH_SOURCE_SETTING \
+        "splashfile=logo.bmp\0" \
+        "splashdevpart=2#bootassets\0" \
+        "splashpos=m,m\0" \
+        BOOTENV
 #endif
 
 #ifdef CONFIG_ARM64
