From ccfee8ca32e32c1127f1a88dc76e6d47e56885a4 Mon Sep 17 00:00:00 2001
From: aravindlal <aravindlal.mh@adlinktech.com>
Date: Wed, 25 Jun 2025 13:03:10 +0530
Subject: [PATCH]  Added TPM2.0 Support

---
 drivers/char/tpm/tpm_tis_i2c.c | 19 ++++++++++++++++++-
 1 file changed, 18 insertions(+), 1 deletion(-)

diff --git a/drivers/char/tpm/tpm_tis_i2c.c b/drivers/char/tpm/tpm_tis_i2c.c
index a897402cc36a..8bd28ff7158f 100644
--- a/drivers/char/tpm/tpm_tis_i2c.c
+++ b/drivers/char/tpm/tpm_tis_i2c.c
@@ -9,10 +9,27 @@
  *
  * It is based on the tpm_tis_spi device driver.
  */
-
+#include <linux/init.h>
+#include <linux/module.h>
+#include <linux/moduleparam.h>
+#include <linux/slab.h>
+#include <linux/interrupt.h>
+#include <linux/wait.h>
+#include <linux/acpi.h>
+#include <linux/freezer.h>
+#include <linux/crc-ccitt.h>
+#include <linux/uaccess.h>
+#include <linux/version.h>
+#include <linux/module.h>
 #include <linux/i2c.h>
 #include <linux/crc-ccitt.h>
 #include "tpm_tis_core.h"
+#include <linux/gpio.h>
+#include <linux/of_irq.h>
+#include <linux/of_gpio.h>
+#include <linux/tpm.h>
+#include <linux/timer.h>
+#include "tpm.h"
 
 /* TPM registers */
 #define TPM_I2C_LOC_SEL 0x00
-- 
2.49.0

