From 13ab2819cf9d41e22019f3be9827068413bb2761 Mon Sep 17 00:00:00 2001
From: aravindlal <aravindlal.mh@adlinktech.com>
Date: Fri, 11 Jul 2025 10:31:54 +0530
Subject: [PATCH] TLV320 audio support

---
 sound/soc/codecs/tlv320aic3x.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/sound/soc/codecs/tlv320aic3x.c b/sound/soc/codecs/tlv320aic3x.c
index 56e795a00e22..731a783263e2 100644
--- a/sound/soc/codecs/tlv320aic3x.c
+++ b/sound/soc/codecs/tlv320aic3x.c
@@ -1581,6 +1581,14 @@ static int aic3x_init(struct snd_soc_component *component)
 	switch (aic3x->model) {
 	case AIC3X_MODEL_3X:
 	case AIC3X_MODEL_33:
+#ifdef CONFIG_ARCH_ADLINKTECH
+               snd_soc_component_write(component, DACL1_2_HPLOUT_VOL, 0x80);
+               snd_soc_component_write(component, DACL1_2_HPROUT_VOL, 0x80);
+               snd_soc_component_write(component, MIC3LR_2_LADC_CTRL, 0x0f);
+               snd_soc_component_write(component, LAGC_CTRL_A, 0x80);
+               snd_soc_component_write(component, RAGC_CTRL_A, 0x80);
+        break;
+#endif
 	case AIC3X_MODEL_3106:
 		aic3x_mono_init(component);
 		break;
-- 
2.49.0

